lib/SSEResponse.js

```diff
@@ -135,28 +135,26 @@ class SSEResponse {
 
 
     /**
-     * Ends the response.
+     * Ends the HTTP response.
      */
     end() {
         if (!this.res) {
             return P.resolve();
         }
+
         if (this._resFinished()) {
             delete this.res;
             return P.resolve();
         }
+
+        const res = this.res;
+        delete this.res;
+
         return new P((resolve, reject) => {
-            const res = this.res;
-            if (this._resFinished()) {
-                delete this.res;
-                return resolve();
-            }
-            delete this.res;
-            res.on('error', reject);
+            res.once('error', reject);
             try {
-                if (res.end(resolve) === false) {
-                    resolve();
-                }
+                res.end();
```

Comment:
Hm... I am wondering why that callback wasn't called... the code returning 'false' was not good, res.end seem to never return 'false'. But it should call the callback..